File: frontend/src/components/common/phoneCountries.ts
export type SupportedPhoneRegion = {
  region: string
  name: string
  dialCode: string
}

export const DEFAULT_PHONE_REGION = 'US'

export const SUPPORTED_PHONE_REGIONS: SupportedPhoneRegion[] = [
  { region: 'US', name: 'United States', dialCode: '+1' },
  { region: 'CA', name: 'Canada', dialCode: '+1' },
  { region: 'PR', name: 'Puerto Rico', dialCode: '+1' },
  { region: 'VG', name: 'British Virgin Islands', dialCode: '+1' },
  { region: 'VI', name: 'U.S. Virgin Islands', dialCode: '+1' },
  { region: 'IN', name: 'India', dialCode: '+91' },
  { region: 'JP', name: 'Japan', dialCode: '+81' },
  { region: 'AT', name: 'Austria', dialCode: '+43' },
  { region: 'BE', name: 'Belgium', dialCode: '+32' },
  { region: 'DK', name: 'Denmark', dialCode: '+45' },
  { region: 'FR', name: 'France', dialCode: '+33' },
  { region: 'DE', name: 'Germany', dialCode: '+49' },
  { region: 'IS', name: 'Iceland', dialCode: '+354' },
  { region: 'IE', name: 'Ireland', dialCode: '+353' },
  { region: 'IM', name: 'Isle of Man', dialCode: '+44' },
  { region: 'IT', name: 'Italy', dialCode: '+39' },
  { region: 'NL', name: 'Netherlands', dialCode: '+31' },
  { region: 'NO', name: 'Norway', dialCode: '+47' },
  { region: 'PT', name: 'Portugal', dialCode: '+351' },
  { region: 'ES', name: 'Spain', dialCode: '+34' },
  { region: 'SE', name: 'Sweden', dialCode: '+46' },
  { region: 'CH', name: 'Switzerland', dialCode: '+41' },
  { region: 'UA', name: 'Ukraine', dialCode: '+380' },
  { region: 'GB', name: 'United Kingdom', dialCode: '+44' },
  { region: 'AR', name: 'Argentina', dialCode: '+54' },
  { region: 'BR', name: 'Brazil', dialCode: '+55' },
  { region: 'CL', name: 'Chile', dialCode: '+56' },
  { region: 'EC', name: 'Ecuador', dialCode: '+593' },
  { region: 'PE', name: 'Peru', dialCode: '+51' },
  { region: 'AU', name: 'Australia', dialCode: '+61' },
  { region: 'CC', name: 'Cocos Islands', dialCode: '+61' },
  { region: 'CX', name: 'Christmas Island', dialCode: '+61' },
  { region: 'NZ', name: 'New Zealand', dialCode: '+64' },
]

export function isSupportedPhoneRegion(region: string): boolean {
  return SUPPORTED_PHONE_REGIONS.some((item) => item.region === region)
}

File: constants/phone_countries.py
"""
Supported phone regions for user SMS numbers.

Use ISO 3166-1 alpha-2 region codes as returned by libphonenumber's
region_code_for_number(). Keep this list in sync with product support.
"""

_SUPPORTED_PHONE_REGION_ROWS = (
    ("US", "United States", "+1"), ("CA", "Canada", "+1"), ("PR", "Puerto Rico", "+1"),
    ("VG", "British Virgin Islands", "+1"), ("VI", "U.S. Virgin Islands", "+1"),
    ("IN", "India", "+91"), ("JP", "Japan", "+81"),
    ("AT", "Austria", "+43"), ("BE", "Belgium", "+32"), ("DK", "Denmark", "+45"),
    ("FR", "France", "+33"), ("DE", "Germany", "+49"), ("IS", "Iceland", "+354"),
    ("IE", "Ireland", "+353"), ("IM", "Isle of Man", "+44"), ("IT", "Italy", "+39"),
    ("NL", "Netherlands", "+31"), ("NO", "Norway", "+47"), ("PT", "Portugal", "+351"),
    ("ES", "Spain", "+34"), ("SE", "Sweden", "+46"), ("CH", "Switzerland", "+41"),
    ("UA", "Ukraine", "+380"), ("GB", "United Kingdom", "+44"),
    ("AR", "Argentina", "+54"), ("BR", "Brazil", "+55"), ("CL", "Chile", "+56"),
    ("EC", "Ecuador", "+593"), ("PE", "Peru", "+51"),
    ("AU", "Australia", "+61"), ("CC", "Cocos Islands", "+61"),
    ("CX", "Christmas Island", "+61"), ("NZ", "New Zealand", "+64"),
)

SUPPORTED_PHONE_REGIONS = tuple(
    {"region": region, "name": name, "dialCode": dial_code}
    for region, name, dial_code in _SUPPORTED_PHONE_REGION_ROWS
)
SUPPORTED_REGION_CODES = {country["region"] for country in SUPPORTED_PHONE_REGIONS}


def serialize_supported_phone_regions() -> list[dict[str, str]]:
    return [dict(country) for country in SUPPORTED_PHONE_REGIONS]
